sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -366,22 +367,34 @@ object BooleanSimplification extends Rule[LogicalPlan] with PredicateHelper {
             // ((c || ...) && (d || ...)) || a || b
```

Comment:
nit: given we are here now, it seems that expression order is not right. Shall we update to

```
a || b || ((c || ...) && (d || ...))
```